### ISSUE TITLE
Missing surface geometries on a wall

### ISSUE THREAD

**Issue by SOCfperrier** [ifcquery/ifcplusplus#106]
Environment : Windows 10, VS2017 x64

This IFC wall is not correctly displayed when loaded from IFCQuery in last available version.
All is ok on Solibri viewer or IFCOpenShell representation meshes.

Find in attachement the IFC file and some screenshots.
Thanks for considering our request.

![solibri](https://user-images.githubusercontent.com/34125283/42022654-1f5c4652-7abe-11e8-88ba-e68512a4f5a6.png)
good

![screen1](https://user-images.githubusercontent.com/34125283/42022662-2612e2ee-7abe-11e8-91d3-ae820fe6aa1f.png)
bad

![screen2](https://user-images.githubusercontent.com/34125283/42022674-2c6c9c98-7abe-11e8-9f7c-8c668b871757.png)
bad too

**Comment by loebl** [comment]
It seems that carve is hitting some floating point border case, so that it errs out and fails to generate a polygon after cutting the hole.

I also encountered problems of that kind - I might have the time to look into this, but I can't guarantee anything. Unfortunately in the current situation you can't do much about it, except trying to modify the dimension a bit and try if that works.

**Comment by rickomax** [comment]
Hi @loebl . Do you accept bounties to work on this issue?

**Comment by loebl** [comment]
No. I have looked into this and this geometry stuff is currently beyond my expertise. And additionally my time is currently bound otherwise, so sorry, I can't help you with that.
